docs: add declarative installer doc (#1054)
redhat-developer/lsp4ij: docs/ContributeToUserDefinedLanguageServer.md

    "default": "sh -c \"typescript-language-server --stdio\"",
    "windows": "typescript-language-server.cmd --stdio"
  },
  "fileTypeMappings": [
    {
      "fileType": {
        "name": "JavaScript"
      },
      "languageId": "javascript"
    },
    {
      "fileType": {
        "name": "JavaScript-React",
        "patterns": [
          "*.jsx"
        ]
      },
      "languageId": "javascriptreact"
    },
    {
      "fileType": {
        "name": "TypeScript",
        "patterns": [
          "*.ts"
        ]
      },
      "languageId": "typescript"
    },
    {
      "fileType": {
        "name": "TypeScript-React",
        "patterns": [
          "*.tsx"
        ]
      },
      "languageId": "typescriptreact"
    }
  ]
}
```

## Installer descriptor

Installer is a JSON descriptor used to:

 * `check` the current installation.
 * `run` the LSP/DAP server installation.

and if the installation is done correctly, `configure` the start command of the server.

It should follow this structure:

```json5
{
  "id": "...",
  "name": "...",
  "check": {
    // Declare check installation tasks here...
  },
  "run": {
    // Declare run installation tasks here...
  }
}
```

> `check` and `run` defines the starting [task](#task-descriptor) to execute and each task can execute other task by using `onFail`, `onSuccess` JSON object.

Reviewer: ```suggestion
`check` and `run` defines the starting [task](#task-descriptor) to execute and each task can execute another task by using the `onFail` and/or `onSuccess` JSON objects.
```

> LSP4IJ provides by default those following tasks:

Reviewer: ```suggestion
LSP4IJ by default provides the following tasks:
```

Author: fixed

* [exec](#task-exec) to execute a command (ex: `npm install typescript-language-server` to install the [typescript-language-server](https://github.com/typescript-language-server/typescript-language-server)).

> * [download](#task-download) to download the language server from a given url according the OS.

Reviewer: ```suggestion
* [download](#task-download) to download the language server from a given url according to the OS.
```

Author: fixed

> * [configureServer](#task-configure-server) to configure the start command by using downloaded server.

Reviewer: ```suggestion
* [configureServer](#task-configure-server) to configure the start command by using the downloaded server.
```

Author: fixed

> If you need other task in your plugin, it is possible to contribute them with `com.redhat.devtools.lsp4ij.installerTaskFactory` extension point.

Reviewer: ```suggestion
If you need other tasks in your plugin, it is possible to contribute them with the `com.redhat.devtools.lsp4ij.installerTaskFactory` extension point.
```

### Task descriptor

> Here the commons task structure:

Reviewer: ```suggestion
Here is the typical task structure:
```

Author: fixed

```json5
{
  "id": "...",
  "name": "...",

>   // custom JSON objects according the task

Reviewer: ```suggestion
  // custom JSON objects according to the task
```

Author: fixed

  "onFail": {
    // Declare a task which must executed when fail
  },
  "onSuccess": {
    // Declare a task which must executed when success
  }
}
```

#### Task exec

> Here a sample which uses `npm` to install [typescript-language-server](https://github.com/typescript-language-server/typescript-language-server):

Reviewer: ```suggestion
Here is a sample which uses `npm` to install [typescript-language-server](https://github.com/typescript-language-server/typescript-language-server):
```

```json
{
  "id": "typescript-language-server",
  "name": "Install and verify TypeScript Language Server",
  "check": {
    "exec": {
      "name": "Check TypeScript Language Server",
      "command": {
        "windows": "where typescript-language-server",
        "default": "which typescript-language-server"
      }
    }
  },
  "run": {
    "exec": {
      "name": "Install TypeScript Language Server Globally",
      "command": {
        "windows": "npm.cmd install -g typescript-language-server",
        "default": "npm install -g typescript-language-server"
      },
      "onSuccess": {
        "configureServer": {
          "name": "Configure TypeScript Language Server command",
          "command": {
            "windows": "typescript-language-server.cmd --stdio",
            "default": "typescript-language-server --stdio"
          },
          "update": true
        }
      }
    }
  }
}
```

#### Task download

> Here a sample which download the JAR at https://oss.sonatype.org/content/repositories/snapshots/io/smartdatalake/sdl-lsp/1.0-SNAPSHOT/sdl-lsp-1.0-20250503.111518-23-jar-with-dependencies.jar

Reviewer: ```suggestion
Here is a sample which downloads the JAR at https://oss.sonatype.org/content/repositories/snapshots/io/smartdatalake/sdl-lsp/1.0-SNAPSHOT/sdl-lsp-1.0-20250503.111518-23-jar-with-dependencies.jar
```

Author: fixed

> and update the start server command to execute the downloaded JAR when language server starts:

Reviewer: ```suggestion
and updates the start server command to execute the downloaded JAR when the language server starts:
```

Author: fixed
